### PR TITLE
Launchrunfarm patient perseverence (local-branch-redux)

### PR DESCRIPTION
**Pull request by timsnyder-siv**
Trying #716 with a local branch to see if the CI works better.

#### UI / API Impact

Adds two parameters to config_runtime.ini to be able to set a timeout for how long to keep looping through AvailabilityZones trying to `launch_instances` and whether `launch_instances` looks at instances that already exist with the specified type and tag and only tries to launch enough to bring up to the requested count or if it always expands the tagged instance pool by the requested count.

#### Verilog / AGFI Compatibility

Doesn't touch verilog or AGFI compatability

### Contributor Checklist
- [X] Did you set dev as the base branch?
- [ ] Did you add Scaladoc to every public function/method?
       Nope because I didn't touch scala bro.
- [X] Did you add at least one test demonstrating the PR?
- [X] Did you delete any extraneous prints/debugging code?
- [X] Did you state the UI / API impact?
- [X] Did you specify the Verilog / AGFI compatibility impact?

- [ ] (If applicable) Did you regenerate and publicly share default AGFIs?

- [ ] (If applicable) Did you mark the PR as "Please Backport"?
- [ ] (On merge) Did you update release notes in the dev-to-master PR ?

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you mark the proper release milestone?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?

### CODE REVIEW

**Comment by davidbiancolin** [comment]
Yes, you'll need push to upstream to get CI to run entirely. We have mergify to automate copying PRs but for frequent contributors with write permissions this is superior.

**Comment by davidbiancolin** [comment]
Interesting, you got transitive approval from #716... GH magic

File: deploy/awstools/awstools.py
Comment (sagark) [inline deploy/awstools/awstools.py:165]: spotinterruptionbehavior

File: deploy/tests/test_amis_snake.json
Comment (sagark) [inline deploy/tests/test_amis_snake.json:1]: is there a script to produce this when the AMI changes? (or really, is there an easy way to get all this information given the AMI id/name)?

File: deploy/awstools/awstools.py
Comment (timsnyder-siv) [reply deploy/awstools/awstools.py:165]: Thanks.  Good catch.
```suggestion
    Using `instancemarket`, `spotinterruptionbehavior` and `spotmaxprice` to define instance market conditions
```

File: scripts/aws-setup.py
Comment (sagark) [inline scripts/aws-setup.py:1]: The wget path in /docs/Initial-Setup/Configuring-Required-Infrastructure-in-Your-AWS-Account.rst

needs to be updated to point to the new location.

File: deploy/tests/test_amis_snake.json
Comment (timsnyder-siv) [reply deploy/tests/test_amis_snake.json:1]: I could/should create one.  I just did `aws ec2 describe-images` to get the AMI json and then munged it into the format that moto expects/accepts.  Before I create one, I'll ask the moto devs if they already have one.

File: scripts/aws-setup.py
Comment (timsnyder-siv) [reply scripts/aws-setup.py:1]: Good catch. I forgot about that expecting to be able to grab the single script to download and run.  Rather than changing the path, should we just check in a symlink so that the wget still works with the old path?  Don't know if anyone has that scripted into anything else.

Comment (sagark) [reply scripts/aws-setup.py:1]: Does wget follow symlinks? If it does, I like that solution.

**Review by sagark — APPROVED** [review]
LGTM modulo wget aws-setup fix!

File: deploy/tests/test_amis_snake.json
Comment (timsnyder-siv) [reply deploy/tests/test_amis_snake.json:1]: Don't even need to ask, just have to look.  oof. https://github.com/spulec/moto/blob/master/scripts/get_amis.py. 

I'll vendor that script into firesim/scripts with smarts to look at the AMI defined in awstools.py and add a python test that fails with a nicer message if the AMI file needs to be updated (otherwise, you'll just get an index error the first time we try to query it and index into the result that doesn't find the AMI we think should always be found).

File: scripts/aws-setup.py
Comment (timsnyder-siv) [reply scripts/aws-setup.py]: GH doesn't serve symlinks as symlinks in the raw.githubusercontent.com URL, it serves them as txt files that have the target of the symlink.  Not what we want.  It looks like those docs may need to be updated to say python3 because I have a feeling that `python` on the FPGA dev AMI may still be python2...  I'm going to double check before updating the docs.

Comment (timsnyder-siv) [reply scripts/aws-setup.py]: Looks like @abejgonzalez already changed the docs on dev to use `./aws-setup.py` to fix the python 2 vs 3 issue.  However, the script had a malformed shebang line and my PR is fixing that.  I'll also update the docs to grab it from the new location with the full-flavor script.

File: deploy/tests/test_amis_snake.json
Comment (timsnyder-siv) [reply deploy/tests/test_amis_snake.json:1]: I'm almost done with creating this script but @davidbiancolin asked me to push it in a follow-on PR so that we can cut the release.  I'll open a follow-on PR later tonight that has it.

Comment (davidbiancolin) [reply deploy/tests/test_amis_snake.json:1]: Thanks Tim.

**Pull request merged**
